### ISSUE TITLE
Strange performance problem on specific configuration

### ISSUE THREAD

**Issue by shodan8192** [nodejs/help#1404]
* **Node.js Version**: 8.11.3, 10.7.0
* **OS**: latest, clean Ubuntu 16.04, Kernel : 4.15.0-29
* **CPU** : Intel i7-3770 (Ivy Bridge)

I'm doing various test of Node.js performance, and I encountered strange problem. On configuration like above, one of my test (Barnsley fern) is running about **4x slower than it is expected**, but with Node.js 6.14.3 is running fast.

On Ubuntu 18.04 with the same kernel and 14.04 with kernel 4.4 there is no problem. Also on newer i7-7700 and an older Core 2 Duo E8400 on Ubuntu 16.04 it's OK.

Please see [files](https://gist.github.com/shodan8192/09ebf36fd68d86ec6e94a2dd5fdf4738) :

- Barnsley fern test code
- Profiler output on Ubuntu 16.04
- Profiler output on Ubuntu 18.04

**Comment by gireeshpunathil** [comment]
thanks, you can actually process the profile output to be more (human) readable via:
$ node --prof-process logfile

good

```
   ticks parent  name
     74   42.0%  LazyCompile: *"barnsley /home/kubuntu/Downloads/barnsley.js:1:80"
     74  100.0%    Script: ~" /home/kubuntu/Downloads/barnsley.js:1:11"
     74  100.0%      LazyCompile: ~"Module._compile module.js:609:37"
     74  100.0%        LazyCompile: ~"Module._extensions..js module.js:661:37"
     74  100.0%          LazyCompile: ~"Module.load module.js:556:33"
     74  100.0%            LazyCompile: ~"tryModuleLoad module.js:502:23"
```
bad

```
   ticks parent  name
    401   44.4%  LazyCompile: *"barnsley /tmp/barnsley.js:1:80"
    401  100.0%    Script: ~" /tmp/barnsley.js:1:11"
    401  100.0%      LazyCompile: ~"Module._compile module.js:609:37"
    401  100.0%        LazyCompile: ~"Module._extensions..js module.js:661:37"
    401  100.0%          LazyCompile: ~"Module.load module.js:556:33"
    401  100.0%            LazyCompile: ~"tryModuleLoad module.js:502:23"
```
So definitely the script took more time to execute. 

I am not sure what would have caused this, but between v6.x and v6.x there would be many changes, and that too positive ones - probably this microbenchmark is is showing a dark area.

Depending on how long the test run and how much you are willing, a series of JIT optimization (+/-) could be applied as well as detailed compilation traces can be collected to examine how the code got transformed. But a more easy thing to do would be to  cc @nodejs/v8 who may be able to provide a better insight.

**Comment by shodan8192** [comment]
Ok, i figured it out. There is a bug in Glibc 2.23, which is shipped with Ubuntu 16.04. It is about AVX-SSE transition penalty, and it reveals on those Intel CPUs generations : Sandy Bridge, Ivy Bridge, Haswell and Broadwell, further reading is [here](https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1663280).

Quick workarounds without upgrading OS are :
- disabling use of AVX in Node.js :
`$ node --no-enable-avx barnsley.js`

- forcing dynamic linker to resolve all the symbols during Node.js startup :
`$ LD_BIND_NOW=1 node barnsley.js`

**Comment by gireeshpunathil** [comment]
great to know, thanks!
